### ISSUE TITLE
Cannot unsubscribe when mqtt client is disconnected

### ISSUE THREAD

**Issue by slgal** [Azure/azure-iot-sdk-java#49]
- **OS and version used:** Windows 10 

- **Java runtime used:** java version "1.8.0_121"

- **SDK version used:**     'com.microsoft.azure.sdk.iot:iot-device-client:1.0.20'

# Description of the issue:
I tried to test Android Sample. I created device id and device key using sample for net. Once I tried to use it for Android I get such error on

```
public void btnReceiveOnClick(View v) throws URISyntaxException, IOException {
        Button button = (Button) v;

        // Comment/uncomment from lines below to use HTTPS or MQTT protocol
        // IotHubClientProtocol protocol = IotHubClientProtocol.HTTPS;
        IotHubClientProtocol protocol = IotHubClientProtocol.MQTT;

        DeviceClient client = new DeviceClient(connString, protocol);

        if (protocol == IotHubClientProtocol.MQTT)
        {
            MessageCallbackMqtt callback = new MessageCallbackMqtt();
            Counter counter = new Counter(0);
            client.setMessageCallback(callback, counter);
        }
        else
        {
            MessageCallback callback = new MessageCallback();
            Counter counter = new Counter(0);
            client.setMessageCallback(callback, counter);
        }

        try {
            client.open();
.....
```

 **client.open();**

Exception while opening IoTHub connection: java.io.IOException: Cannot unsubscribe when mqtt client is disconnected

Can you please explain what can cause that error?

**Comment by JMayrbaeurl** [comment]
This is related to Issue #47 . The causing exception is hidden and stored as root cause in the IOException. Check with the debugger.

**Comment by prmathur-microsoft** [comment]
Thanks for the input. As you mentioned, this will be fixed along with #47

**Comment by slgal** [comment]
I debugged it and find what cause it. I forgot to add:
  ```<uses-permission android:name="android.permission.INTERNET"/>```
Now it's works fine. Thanks a lot.
